src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -309,25 +302,52 @@ private static string GetProjectOutputPath(string projectDirectory, string msbui
 
         internal static DependencyGraphSpec CreateMinimalDependencyGraphSpec(string projectPath, string outputPath)
         {
-            var packageSpec = new PackageSpec();
-            packageSpec.FilePath = projectPath;
-            packageSpec.RestoreMetadata = new ProjectRestoreMetadata();
-            packageSpec.RestoreMetadata.ProjectUniqueName = projectPath;
-            packageSpec.RestoreMetadata.ProjectStyle = ProjectStyle.PackageReference;
-            packageSpec.RestoreMetadata.ProjectPath = projectPath;
-            packageSpec.RestoreMetadata.OutputPath = outputPath;
-            packageSpec.RestoreMetadata.CacheFilePath = Path.Combine(outputPath, "project.nuget.cache");
+            var packageSpec = new PackageSpec
+            {
+                FilePath = projectPath,
+                RestoreMetadata = new ProjectRestoreMetadata()
+                {
+                    ProjectUniqueName = projectPath,
+                    ProjectStyle = ProjectStyle.PackageReference,
+                    ProjectPath = projectPath,
+                    OutputPath = outputPath,
+                    CacheFilePath = NoOpRestoreUtilities.GetProjectCacheFilePath(outputPath),
+                }
+            };
 
             var dgSpec = new DependencyGraphSpec();
             dgSpec.AddProject(packageSpec);
 
             return dgSpec;
         }
 
-        public Task RegisterRestoreInfoSourceAsync(IVsProjectRestoreInfoSource restoreInfoSource, CancellationToken cancellationToken)
+        public async Task RegisterRestoreInfoSourceAsync(IVsProjectRestoreInfoSource restoreInfoSource, CancellationToken token)
         {
-            return Task.CompletedTask;
-            // TODO: https://github.com/NuGet/Home/issues/10678
+            if (restoreInfoSource == null)
```

Comment:
`is null`?

Reply:
Me neither, but I was given PR feedback a few months ago to explicitly change mine to `is null` for all new code.

Reply:
I see. 
Seems like a very irrelevant thing to be honest. 
I'd advocate for consistency in the file. 
I've also merged plenty of code over the last few months, with `== null` checks :D